omod/src/main/web/dashboardwidgets/programstatus/programstatus.controller.js

```diff
@@ -466,8 +463,42 @@ export default class ProgramStatusController {
             this.patientProgram.states = this.$filter('filter')(this.patientProgram.states, (state) => {
                 return !state.voided
             }, true);
-            this.patientProgram.states = this.$filter('orderBy')(this.patientProgram.states, (state) => {
-                return new Date(state.startDate);
+            this.patientProgram.states = this.$filter('orderBy')(this.patientProgram.states, null, false, function(state1, state2)
+             {
+            	var result = 0;
+            	
+            	//console.log(state1, state2);
+            	
+            	//if the start date of 1 is earlier, is occurs before 
+            	if(state1.value.startDate < state2.value.startDate) {
+            		//console.log("based on startDate", state1.value.startDate, "is before", state2.value.startDate);
+            		result = -1;
+            	//later it occurs after
+            	} else if (state1.value.startDate > state2.value.startDate) {
+            		result = 1;
+            	} else {
+            	
+	            	//if the end date is not set, or end date is after, the state occurred before
+	            	if(state1.value.endDate < state2.value.endDate || state2.value.endDate == null ) {
```

Comment:
Is this null-safe?  Should the null checks happen first?

Reply:
I'll switch it to the null check first. `Date() < null`  and `Date() > null` are both `false` in Chrome, but I don't know if that behavior is specifically required by js/EMCAScript.